build/devices/linemb/xsProj-glib/main.c

```diff
@@ -0,0 +1,271 @@
+	}
+}
+
+static void fatal_error_exit() {
+  void *buffer[50];
+  int nptrs = backtrace(buffer, 50);
+  char **strings = backtrace_symbols(buffer, nptrs);
+  printf("==== Native stack trace: =====\n");
+  for (int i = 0; i < nptrs; i++)
+  {
+    printf("%s\n", strings[i]);
+  }
+  free(strings);
+  printf("==============================\n");
+  printf(FATAL_ERROR_TOKEN);
+  
+  signal(SIGQUIT, SIG_DFL);
+  exit(1);
+}
+
+static void fatal_error_handler(int signum) {
+  const char* signame = "UNKNOWN";
+  switch(signum) {
+    case SIGABRT: signame = "SIGABRT"; break;
+    case SIGFPE:  signame = "SIGFPE"; break;
+    case SIGILL:  signame = "SIGILL"; break;
+    case SIGINT:  signame = "SIGINT"; break;
+    case SIGQUIT: signame = "SIGQUIT"; break;
+    case SIGSEGV: signame = "SIGSEGV"; break;
+    case SIGTERM: signame = "SIGTERM"; break;
+    case SIGBUS:  signame = "SIGBUS"; break;
+    case SIGPIPE: signame = "SIGPIPE"; break;
+  }
+  
+  printf("!!!! Signal %s (%d) caught. Stack trace:\n", signame, signum);
+  // Reset signal handler to allow default behavior to terminate program
+  signal(signum, SIG_DFL);
+  fatal_error_exit();  
+}
+
+// Modified: Unified signal handling function
+static void signal_handler(int signum) {
+  fatal_error_handler(signum);
+}
+
+static void timeout_handler(int signum) {
+  if (gxMachine) {
+    printf("!!!! Execution timeout !!!!\n");
+    dump_js_stack(gxMachine);
+    fatal_error_exit();
+  }
+}
+
+int main(int argc, char *argv[]) {
+  // Use setvbuf to disable buffering
+  setvbuf(stdout, NULL, _IONBF, 0);
+  setvbuf(stderr, NULL, _IONBF, 0);
+
+  // Register all signals to be captured
+  signal(SIGABRT, signal_handler);
+  signal(SIGFPE, signal_handler);
+  signal(SIGILL, signal_handler);
+  signal(SIGQUIT, signal_handler);
+  signal(SIGSEGV, signal_handler);
+  signal(SIGTERM, signal_handler);
+  signal(SIGBUS, signal_handler);
+  signal(SIGPIPE, signal_handler);
+  signal(SIGALRM, timeout_handler); // Used for timeout
+
+  int error = 0;
+  static txMachine _root;
+  txMachine *the = &_root;
+  txPreparation *preparation = xsPreparation();
+
+  c_memset(the, 0, sizeof(txMachine));
+  the->preparation = preparation;
+  the->keyArray = preparation->keys;
+  the->keyCount = (txID)preparation->keyCount + (txID)preparation->creation.initialKeyCount;
+  the->keyIndex = (txID)preparation->keyCount;
+  the->nameModulo = preparation->nameModulo;
+  the->nameTable = preparation->names;
+  the->symbolModulo = preparation->symbolModulo;
+  the->symbolTable = preparation->symbols;
+
+  the->stack = &preparation->stack[0];
+  the->stackBottom = &preparation->stack[0];
+  the->stackTop = &preparation->stack[preparation->stackCount];
+
+  the->firstHeap = &preparation->heap[0];
+  the->freeHeap = &preparation->heap[preparation->heapCount - 1];
+  the->aliasCount = (txID)preparation->aliasCount;
+
+  setvbuf(stdout, NULL, _IONBF, 0);
+  the = fxCloneMachine(&preparation->creation, the, "linemb", NULL);
```

Comment:
This is more-or-less `xsPrepareMachine` (implemented in [`fxPrepareMachine`](https://github.com/Moddable-OpenSource/moddable/blob/2945c5d0f2d37e534a3d09729f25e576b0561676/xs/sources/xsAPI.c#L1736)). If possible, it would be more maintainable to use the `xsPrepareMachine`.